Tests/CognitoIdentityDemoTests/CognitoIdentityDemoTests.swift

```diff
@@ -19,18 +19,26 @@ final class CognitoIdentityDemoTests: XCTestCase {
     /// **Test:** Attempt to find an identity pool that doesn't exist. If no error occurs, the test
     /// fails.
     func testFindNonexistent() {
-        let poolID = identityTester?.getIdentityPoolID(name: "BogusPoolIsBogus", createIfMissing: false)
-        XCTAssertNil(poolID, "Found identity pool that does not exist")
+        identityTester?.getIdentityPoolID(name: "BogusPoolIsBogus", createIfMissing: false) { poolID in
+            XCTAssertNil(poolID, "Found identity pool that does not exist")
+        }
     }
     
     /// **Test:** Create (or locate, if it already exists) an identity pool. Then try to find it
     /// a second time.. Make sure the returned IDs match. If not, the test fails.
     func testCreateThenFind() {
-        let firstPoolID = identityTester!.getIdentityPoolID(name: "testCreateThenFind", createIfMissing: true)
-        XCTAssertNotNil(firstPoolID, "Unable to create or obtain test pool")
+        var firstPoolID: String?
+        var secondPoolID: String?
```

Comment:
no need for these. use XCTExpectation to fulfill async expectations in tests. :)

Reply:
How can you compare the result of two tests without storing the value of at least the first result until the second result arrives?

Reply:
the calls should be wrapped in each other, welcome to callback hell. you cant really call the second one until the first has completed. and since you wont know if its completed unless you make the call to the second inside the completion handler of the first.
```
        identityTester?.getIdentityPoolID(name: "testCreateThenFind", createIfMissing: true) { poolID in
            let firstPoolId = poolID
            XCTAssertNotNil(firstPoolId, "Unable to create or obtain test pool")
              identityTester?.getIdentityPoolID(name: "testCreateThenFind", createIfMissing: false) { poolID in
                  let secondPoolID = poolID
                  XCTAssertNotNil(secondPoolID, "Unable to find test pool")
                  XCTAssertTrue(firstPoolID == secondPoolID, "Found pool ID doesn't match created pool ID")
              }
        }


```